src/classes/storage.ts

```diff
@@ -84,4 +100,155 @@ export class Storage {
 
     this.groundPlaces[placeGpuid] = newGroundPlace;
   }
+
+  /**
+   * @description Allows you to filter the ground places according to criteria such as name, gpuid, etc.
+   * @param {AutoCompleteFilters[]|undefined} filters - The filters you want to be applied on the search.
+   * @returns {AutoComplete[]}
+   */
+  public filterGroundPlaces(filters?: AutoCompleteFilters[]): AutoComplete[] {
+    const groundPlacesAutocomplete: AutoComplete[] = this.parseGroundPlaces();
+
+    // Returns earlier the Ground places if there is no filters to use.
+    if (!filters || !filters.length) return groundPlacesAutocomplete;
+
+    const filteredByGroup: AutoComplete[] = this.filterGroundPlacesByGroup(
+      groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.STOP_GROUP),
+    );
+
+    const filteredByCluster: AutoComplete[] = this.filterGroundPlacesByCluster(
+      groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.STOP_CLUSTER),
+    );
+
+    const filteredByType: AutoComplete[] = [...filteredByGroup, ...filteredByCluster];
+
+    const filteredBySegmentProviderStop: AutoComplete[] = this.filterGroundPlacesBySegmentProvider(
+      // Filtering by segment provider must be made on the whole results previously filtered.
+      // We checks here if this filter must be applied on results already filtered or not.
+      filteredByType.length ? filteredByType : groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.SEGMENT_PROVIDER_STOP),
+    );
+
+    const filteredByService: AutoComplete[] = this.filterGroundPlacesByServiced(
+      filteredBySegmentProviderStop,
+      filters.includes(AutoCompleteFilters.SERVICED),
+    );
+
+    return filteredByService;
+  }
+
+  /**
+   * @description Search through ground places with specific words.
+   * @param {AutoComplete[]} groundPlaces - The places on which the search is used.
+   * @param {string} search - The search to find the requested places.
+   * @returns {AutoComplete[]}
+   */
+  public searchGroundPlaces(groundPlaces: AutoComplete[], search: string): AutoComplete[] {
+    return groundPlaces.filter((place) => {
+      // Method toLowerCase() is used because the includes() method is case sensitive
+      const currentSearch = search.toLowerCase();
+
+      /* Search by unique name. This search only concern StopCluster. */
+      if (place.type === GroundPlaceType.CLUSTER && place.unique_name.toLowerCase().includes(currentSearch)) {
```

Comment:
Same your if is not supposed to be there. 
You should : 
1. search 
2. filter

(saearch & filter are simple lodash function)

Reply:
I didn't quite understand the remark about "your if is not supposed to be there". 

This function is used by the autocomplete method on the GroundPlacesController class to search trough ground places with specific words. 

This `if` is used firstly to check if the phrase that the user type match an `unique_name`, `unique_name` is a usable parameter  only on `StopCluster`, this condition means that we search if there are place with type `cluster` matching with `unique_name` parameter. If it is, we return earlier the place, if not, we checking with the next criteria such as gpuid, or name (that are used both on `StopGroup` and `StopCluster`).

Compared to `lodash`, its use had a real interest when we were not yet in `ES6`, now all the native JavaScript functions allow to do what `lodash` proposes. 

At the performance level, many benchmarks have been made and it has been shown that using `lodash` functions is not as performant as `JS vanilla`.

I will be able to use the same functions as lodash, but I will end up producing a similar result in terms of code logic with a slower execution speed.

What is you opinion on this @rimiti @theo-le-donne

Reply:
Don't care about opti for now. Think simple. 
Lodash if you like, vanilla if you prefer.

Reply:
because right it's neither optimized nor simple

Reply:
no need for lodash just for `filter`. I think @Simobain just misunderstood the if the ensure the `place.unique_name` exists (only on GroundPlaceType.CLUSTER) . But you could just check if the property exists like `if (place.unique_name && place.unique_name.toLowerCase().includes(currentSearch)) {...}` ou `place.unique_name?.toLowerCase()`